pkg/phases/os.go

```diff
@@ -20,6 +20,15 @@ type OS interface {
 
 	//DetectAtRuntime will detect if it is compatible with the current running OS
 	DetectAtRuntime() bool
+
+        // Returns the names of the kernel and kernel header packages
+        GetKernelPackageNames(version string) (string, string)
+
+        // Calls the version specific grub config logic
+        SetDefaultKernel(version string) types.Commands
+
+        // Check whether defaul kernel matches specific version
+        VerifyDefaultKernel(version string) bool
```

Comment:
The verification logic should move into the Verify function in kernel.go, e.g. `sys.Kernel == os.ReadDefaultKernel()`

Reply:
I'm not 100% sure what you're asking here.  ReadDefaultKernel itself does not indicate whether the change was implemented correctly, it just returns the value inside grub.  The mapping between the input string and the output of ReadDefaultKernel is OS specific (especially for grubconf systems), so I kept the logic to verify that mapping in the OSes.  So sys.Kernel.Verify calls os.VerifyDefaultKernel which calls and compares os.ReadDefaultKernel

Reply:
>  mapping between the input string and the output of ReadDefaultKernel is OS-specific 

It might make sense to move that mapping into ReadDefaultKernel or another OS.method - The OS package should be a pure abstraction and shouldn't contain any verification logic - at the moment the sys package is also a little "leaky" it contains both apply/verification logic and some OS differentiation, if we move verification into os, then os becomes "leaky" as well,

Reply:
So something like sys.Kernel.Verify checks GetDefaultKernel == ReadDefaultKernel ?

Reply:
Or more concretely `phases/kernel.go:Verify` would check `sys.Kernel == os.ReadDefaultKernel()`